core/Pos/Core/Address.hs

```diff
@@ -346,6 +347,46 @@ deriveLvl2KeyPair ibea scp passphrase wsKey accountIndex addressIndex = do
     -- We don't need to check passphrase twice
     createHDAddressH ibea (ShouldCheckPassphrase False) passphrase hdPass wKey [accountIndex] addressIndex
 
+-- | Derives a HD wallet address and private key from a root key and a path
+-- which is 3 levels deep (i, j, k).
+deriveLvl3KeyPair
+    :: Bi Address'
+    => IsBootstrapEraAddr
+    -> ShouldCheckPassphrase
+    -> PassPhrase
+    -> EncryptedSecretKey
+    -> (Word32, Word32, Word32)
```

Comment:
Just a thought: was about adding a comment specifying what those three `Word32` are about? Ideally eventually we will type-synonym those to something like `DerivationIndex`, but for now what about:

```
-> (Word32, Word32, Word32)
-- ^ (account index, chain index, address index)
```

The name `chain index` was stolen from [here](https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki#master-key-generation). Maybe that's not a good idea, but I thought speaking as much as possible the BIP-32 vocabulary could be a good thing 😉

Reply:
Updated in latest commit.